### PR TITLE
[move-ide] Fixes auto-completion for fields

### PR DESCRIPTION
**Pull request by awelc**
## Description 

This PR fixes field auto-completion in two ways:
- adds support for struct fields (previously only variant fields were being auto-completed)
- named fields are now listed in their definition order (which should typically be what a developer wants)
- finesses auto-completion formatting (previously all fields where inserted on a single line, and now named fields are inserted on separate lines if there are more than two fields)
- adds a no-field auto-completion option (for when a struct or variant is used as a type and not in pack/unpack context)

## Test plan 

All tests must pass

### CODE REVIEW

File: external-crates/move/crates/move-analyzer/src/completion.rs [hunk 1]
```diff
@@ -194,50 +195,71 @@ fn get_cursor_token(buffer: &str, position: &Position) -> Option<Tok> {
     }
 }
 
-/// Handle context-specific auto-completion requests with lbrace (`{`) trigger character.
```
Comment (awelc) [inline external-crates/move/crates/move-analyzer/src/completion.rs:197]: This is an unrelated refactoring that I sneaked in to get rid of `typed_ast` in `Symbols` (it was annoying that it was only used in this one place and it wasn't even needed as the same info is encoded elsewhere)

File: external-crates/move/crates/move-analyzer/src/analysis/parsing_analysis.rs [hunk 2]
```diff
@@ -802,47 +803,3 @@ fn parsing_mod_ident_to_map_key(
     )
     .to_string()
 }
-
```
Comment (awelc) [inline external-crates/move/crates/move-analyzer/src/analysis/parsing_analysis.rs:805]: This was just moved around without change so that it can be used in `completions.rs`

File: external-crates/move/crates/move-analyzer/src/symbols.rs [hunk 3]
```diff
@@ -1993,6 +2079,50 @@ pub fn expansion_mod_ident_to_map_key(mod_ident: &E::ModuleIdent_) -> String {
...
+
+/// Produces module ident string of the form pkg::module to be used as a map key.
+/// It's important that these are consistent between parsing AST and typed AST.
+pub fn parsing_mod_def_to_map_key(
+    pkg_addresses: &NamedAddressMap,
+    mod_def: &P::ModuleDefinition,
+) -> Option<String> {
+    // we assume that modules are declared using the PkgName::ModName pattern (which seems to be the
+    // standard practice) and while Move allows other ways of defining modules (i.e., with address
+    // preceding a sequence of modules), this method is now deprecated.
```
Comment (rvantonder) [inline external-crates/move/crates/move-analyzer/src/symbols.rs:2117]: 👍

**Review by rvantonder — APPROVED** [review]
Looks good to go!
